### PR TITLE
[ISSUE #26764] support brute force multiline json objects for JSONL

### PR DESCRIPTION
**Pull request by maxi297**
## What
Adresses https://github.com/airbytehq/airbyte/issues/26764

As mentioned in the ticket:
* Specifying `newlines_in_values` does not change the behavior i.e. having this True or False still allows for multiline JSON and therefore, all JSONL files synced in production could have been considered "newlines_in_values"
* The drawback `block_size` issues using pyarrow seems too annoying for now as it broke stream before
* We agreed on supporting multiline jsonl by brute force in the CDK and adding log to see if users actually have this. We will add a reminder to search for this log after the release. Unfortunately, we will only know this once V4 is installed though. I've created this reminder on August 21st. Ping me if you want to be invited to the event

## How
If a line is not parsable, accumulate the content of this line and parse using the following line. Rince and repeat until JSON parsing works

### CODE REVIEW

File: airbyte-cdk/python/unit_tests/sources/file_based/file_types/test_jsonl_parser.py [hunk 1]
```diff
@@ -33,3 +35,66 @@ def test_type_mapping(record: Dict[str, Any], expected_schema: Mapping[str, str]
...
+def test_given_one_json_per_line_when_parse_records_then_return_records() -> None:
+    stream_reader = MagicMock()
+    stream_reader.open_file.return_value.__enter__.return_value = JSONL_CONTENT_WITHOUT_MULTILINE_JSON_OBJECTS
+
+    records = list(JsonlParser().parse_records(Mock(), Mock(), stream_reader, Mock()))
+
+    assert records == [{"a": 1, "b": "1"}, {"a": 2, "b": "2"}]
+
+
+def test_given_multiline_json_object_when_parse_records_then_return_records() -> None:
```
Comment (girarda) [inline airbyte-cdk/python/unit_tests/sources/file_based/file_types/test_jsonl_parser.py:74]: this test should verify we don't log a warning

File: airbyte-cdk/python/airbyte_cdk/sources/file_based/file_types/jsonl_parser.py [hunk 2]
```diff
@@ -16,6 +17,7 @@
 class JsonlParser(FileTypeParser):
 
     MAX_BYTES_PER_FILE_FOR_SCHEMA_INFERENCE = 1_000_000
+    _WITH_READ_LIMIT = True
```
Comment (brianjlai) [inline airbyte-cdk/python/airbyte_cdk/sources/file_based/file_types/jsonl_parser.py:20]: Do you plan on using this constant more? It feels unnecessary if we're only using it in one place when infer_schema calls `infer_jsonl_entries()`

File: airbyte-cdk/python/unit_tests/sources/file_based/file_types/test_jsonl_parser.py (hunk 1 above)
Comment (maxi297) [reply airbyte-cdk/python/unit_tests/sources/file_based/file_types/test_jsonl_parser.py]: Did you mean that we should log a warning? `test_given_multiline_json_object_when_parse_records_then_log_once_one_record_yielded` validates that. I can check that `test_given_one_json_per_line_when_parse_records_then_return_records` does not log a warning for extra security

Comment (girarda) [reply airbyte-cdk/python/unit_tests/sources/file_based/file_types/test_jsonl_parser.py]: I think I derped and commented on the wrong line. +1 for verifying we do not log a warning when only parsing single line records

**Review by girarda — APPROVED** [review]
:shipit:

**Pull request merged**